File: src/test/java/net/igsoft/sdi/testclasses/ECreator.java
package net.igsoft.sdi.testclasses;

import net.igsoft.sdi.creator.CreatorBase;
import net.igsoft.sdi.engine.InstanceProvider;
import net.igsoft.sdi.parameter.LaunchType;

public class ECreator extends CreatorBase<EClass, LaunchType> {
    @Override
    public EClass create(InstanceProvider instanceProvider, LaunchType launchType) {
        return new EClass(instanceProvider.getOrCreate(Stepper.class));
    }
}

File: src/test/scala/net/igsoft/sdi/scala/testclasses/ECreator.scala
package net.igsoft.sdi.scala.testclasses

import net.igsoft.sdi.{Creator, InstanceCreator, LaunchType}

class ECreator extends Creator[E, LaunchType] {
  override def create(instanceCreator: InstanceCreator, launchType: LaunchType) = new E(instanceCreator
                                                                                        .getOrCreate(classOf[Stepper]))
}
